### PR TITLE
ref(ts): Add typings for reflux

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Need these to be able to properly type hook store

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/withOrganizations.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const withOrganizations = <P extends InjectedOrganizationsProps>(
     State
   >({
     displayName: `withOrganizations(${getDisplayName(WrappedComponent)})`,
-    mixins: [Reflux.connect(OrganizationsStore, 'organizations')],
+    mixins: [Reflux.connect(OrganizationsStore, 'organizations') as any],
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/withOrganizations.tsx]: Why do these need to cast to any?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: The reflux typings aren't very good, and apparently `Reflux.connect` [returns `void`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/reflux/v0/index.d.ts#L55), which is not compatible with [`mixins`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L710).

Comment (dashed) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: Yea they're not good. I feel like we can roll out our own definitions that are better.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: meh, I would rather just get rid of reflux at some point. I just wanted the typings for the interface

Comment (markstory) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: Should we update the typings?

Comment (markstory) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: Yeah, that would let us correct the typings upstream.

**Comment by EvanPurkhiser** [comment]
Would like to merge this in ASAP since #15145 needs it.

File: src/sentry/static/sentry/app/utils/withOrganizations.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: +1 on not investing time on typing reflux

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/utils/withOrganizations.tsx]: What do I need to do to get a +1 here :)

**Review by dashed — APPROVED** [review]
Lets get some reflux types 👍

**Pull request merged**